src/CUDAKernels.jl

```diff
@@ -243,4 +243,10 @@ function KA.priority!(::CUDABackend, prio::Symbol)
     return nothing
 end
 
+KA.supports_enzyme(::CUDABackend) = true
+function KA.__fake_compiler_job(::CUDABackend)
+    mi = CUDA.methodinstance(typeof(()->return), Tuple{})
+    return CUDA.CompilerJob(mi, CUDA.compiler_config(CUDA.device()))
+end
```

Comment:
This is very sketchy...

Reply:
So the core challenge we have is that we need to allocate memory of a certain type for the tape. This memory allocation needs to occur on the outside and then be passed into the kernel.

What we came up with was a reflection function `tape_type` that given a compilation job returns the element type of the array to allocate.

The crux is that we can't use the host job for the reflection since in the end this is a deferred compilation. This requires us taking into account the CUDA method table.

So for this reflection we need the parent job. Not the real one as long as the method table matches, also the parent kernel will take the allocated array as an argument so we can't even construct it yet.